### PR TITLE
Exception handling for index creation/deletion

### PR DESCRIPTION
**Pull request by DanJRWalsh**
Added missing exception handling to `CreateIndex`, `CreateIndexAsync` and `DropIndexAsync`

### CODE REVIEW

File: src/Redis.OM/RediSearchCommands.cs [hunk 1]
```diff
@@ -130,6 +142,31 @@ public static bool DropIndexAndAssociatedRecords(this IRedisConnection connectio
             }
         }
 
+        /// <summary>
+        /// Deletes an index. And drops associated records.
+        /// </summary>
+        /// <param name="connection">the connection.</param>
+        /// <param name="type">the type to drop the index for.</param>
+        /// <returns>whether the index was dropped or not.</returns>
+        public static async Task<bool> DropIndexAndAssociatedRecordsAsync(this IRedisConnection connection, Type type)
```
Comment (slorello89) [inline src/Redis.OM/RediSearchCommands.cs:151]: I'm thinking we really don't need the `DropIndexAndAssociatedRecordsAsync` method as this is really something you shouldn't be doing in a situation where sync/async is of major importance 😂.

**Review by slorello89 — CHANGES_REQUESTED** [review]
Thanks for submitting this PR, yeah I'm good with fixing DropIndexAsync as it's going to toss an exception out as if the index already exists, which sort of defeats the purpose of the boolean. 

Please add a functional test to validte this fix, should be fairly trivial to add.

However with the `DropIndexAndAsociatedRecords` I'm skeptical of adding an async version of that method, that method (and the index creation/deletion methods in general) are methods that really shouldn't be run in a situation where sync vs async code is important, so let's pull out the `DropIndexAndAssociatedRecordsAsync` method

File: src/Redis.OM/RediSearchCommands.cs (hunk 1 above)
Comment (DanJRWalsh) [reply src/Redis.OM/RediSearchCommands.cs:151]: @slorello89 That's fair but with that logic should we not remove `DropIndexAsync` then? I was adding it more for consistency than anything.

Comment (slorello89) [reply src/Redis.OM/RediSearchCommands.cs]: Hi @DanJRWalsh Yeah, I think that's fair please by all means go ahead and add that.

**Comment by DanJRWalsh** [comment]
@slorello89 Let me know if it needs anything else

**Review by slorello89 — APPROVED** [review]
LGTM 👍

**Pull request merged**